builder/vagrant/step_initialize_vagrant.go

```diff
@@ -0,0 +1,136 @@
+package vagrant
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepInitializeVagrant struct {
+	BoxName      string
+	BoxVersion   string
+	Minimal      bool
+	Template     string
+	SourceBox    string
+	OutputDir    string
+	SyncedFolder string
+	GlobalID     string
+}
+
+var DEFAULT_TEMPLATE = `Vagrant.configure("2") do |config|
+  config.vm.box = "{{.BoxName}}"
+  {{ if ne .SyncedFolder "" -}}
+  		config.vm.synced_folder "{{.SyncedFolder}}", "/vagrant"
+  {{- else -}}
+  		config.vm.synced_folder ".", "/vagrant", disabled: true
+  {{- end}}
+end`
+
+type VagrantfileOptions struct {
+	SyncedFolder string
+	BoxName      string
+}
+
+func (s *StepInitializeVagrant) createInitializeCommand() (string, error) {
```

Comment:
I think this func could be doing too much ; I think it should probably just return a `*template.Template`.

Suggestion:
```suggestion
func (s *StepInitializeVagrant) getVagrantfileTemplate() (*template.Template, error) {
```

Reply:
I'm not sure I understand your reasoning behind why I should be returning a pointer to a template rather than the file the template gets written to; if I don't execute the template in this function that it really isn't doing anything, and if I do execute the template in this function then the template itself is not important; it's path is.

Reply:
Sorry I skipped a few steps here...
So, because I think the we should [only create Vagrantfile and not init](https://github.com/hashicorp/packer/pull/7221/files/e8c655bf157867f8403776ce00e8a17ebd471dfa#r254039153). ( if I am correct )
I think that there should be a `prepVagrantFile`  that will call `getVagrantfileTemplate`, ex:

```go
func (s *StepInitializeVagrant) prepVagrantFile() error
	vgtpl, err := s.getVagrantfileTemplate()
	if err != nil {
		return err
	}
	vf, err := os.Create(filepath.Join(s.OutputDir, "Vagrantfile"))
	if err != nil {
		return err
	}
	return vgtpl.Execute(vf, VagrantfileOptions{
		SyncedFolder: s.SyncedFolder,
		BoxName:      s.SourceBox,
	})
```
This makes the func very easy to read.

PS: I didn't spent much time thinking on the func names, may be there are better ones !

Reply:
Ah, now I understand. That does make sense.